src/claim_proofs/claim_proofs.rs

```diff
@@ -57,22 +56,55 @@ lazy_static! {
     static ref SIG_CTXT: SigningContext = signing_context(SIGNING_CTX);
 }
 
-/// The data needed to generate a CDD ID
+/// Create a scalar from a slice of data.
+fn slice_to_scalar(data: &[u8]) -> Scalar {
+    use blake2::{Blake2b, Digest};
+    let mut hash = [0u8; 64];
+    hash.copy_from_slice(
+        Blake2b::default()
+            .chain(data.as_ref())
+            .finalize()
+            .as_slice(),
+    );
+    Scalar::from_bytes_mod_order_wide(&hash)
```

Comment:
```suggestion
    let hash = Blake2b::digest(data);
    Scalar::from_bytes_mod_order_wide(hash.as_ref())
```
It's just a utility function but it does the same thing.

Reply:
`Scalar::from_bytes_mod_order_wide()` expects a slice of exactly 64 bytes, that's why we have to copy the hash into a slice. I agree that in this case calling `digest()` is cleaner than `chain()`.